pkg/login/social/connectors/google_oauth.go

```diff
@@ -71,6 +71,27 @@ func (s *SocialGoogle) Validate(ctx context.Context, settings ssoModels.SSOSetti
 	return nil
 }
 
+func (s *SocialGoogle) Reload(ctx context.Context, settings ssoModels.SSOSettings) error {
+	info, err := CreateOAuthInfoFromKeyValues(settings.Settings)
+	if err != nil {
+		return fmt.Errorf("SSO settings map cannot be converted to OAuthInfo: %v", err)
+	}
+
+	config := createOAuthConfig(info, s.cfg, social.GoogleProviderName)
+
+	if strings.HasPrefix(info.ApiUrl, legacyAPIURL) {
```

Comment:
I wonder if this should be part _Validate_ instead?

Reply:
Yes (I have a separate issue to review the validation), but it should be part of the Reload as well, because the Reload function will be called after the DB settings has been removed and in that case the data will come from the `system` config.